safe/gui/tools/people_in_buildings_dialog.py

```diff
@@ -0,0 +1,380 @@
+            return
+        field_names = self._get_field_names(layer)
+        self._update_combobox(self.usageColumnComboBox, field_names)
+        self._update_combobox(self.levelsColumnComboBox, field_names)
+
+    def handle_census_layer(self, layer_name):
+        """Hander for change of population layer selection event.
+
+        :param layer_name: The name of the layer that was selected.
+        :type layer_name: basestring
+        """
+        layer = self._select_layer_by_name(layer_name)
+        if layer is None:
+            return
+        field_names = self._get_field_names(layer)
+        self._update_combobox(self.populationCountComboBox, field_names)
+
+    @staticmethod
+    def _update_combobox(combobox, options):
+        """Shorthand for clearing and loading options (this happens a lot)
+        """
+        combobox.clear()
+        combobox.addItems(options)
+
+    @staticmethod
+    def _add_attribute_to_layer(layer, attribute_name, data_type):
+        """Add a new attribute to an existing layer.
+
+        :param layer: The layer to be altered.
+        :type layer: VectorLayer
+
+        :param attribute_name: The name of the new attribute.
+        :type attribute_name: basestring
+
+        :param data_type: The type of the new attribute.
+        :type data_type: QVariant.Double, QVariant.Int
+
+        :returns: The success state of the operation.
+        :rtype: bool
+        """
+        provider = layer.dataProvider()
+        # Check if attribute is already there, return "-1" if not
+        ind = provider.fieldNameIndex(attribute_name)
+        if ind != -1:
+            return False
+        field = QgsField(attribute_name, data_type)
+        result = provider.addAttributes([field])
+        return result
+
+    def add_population_attribute(self, layer):
+        """Add people estimate columns/attributes to the output layer.
+
+        :param layer: The layer that the population attribute should be added.
+        :type layer: VectorLayer
+        """
+        attributes = {
+            People_Calculated: QVariant.Double,
+            People_Rounded: QVariant.Int
+        }
+        for attribute_name, data_type in attributes.items():
+            result = self._add_attribute_to_layer(
+                layer, attribute_name, data_type)
+            if result:
+                layer.updateFields()
+
+    @staticmethod
+    def _get_attributes(feature, field_names):
+        """Get the attribute dict based on the feature and the field names.
+
+        :param feature: The feature to be investigated.
+        :type feature:
+
+        :param field_names: The field names list.
+        :type field_names: list
+
+        :return: A lookup of the field values by field name.
+        :rtype: dict
+        """
+        attributes = feature.attributes()
+        attribute_dict = dict(zip(field_names, attributes))
+        return attribute_dict
+
+    @staticmethod
+    def _overlapping_area(feature, layer):
```

Comment:
I thought we had said we would use building centroids for determining if a building is flooded etc. so that we can prevent double counting using overlaps?